### PR TITLE
Update msal dependency

### PR DESCRIPTION
**Pull request by g2vinay**

### CODE REVIEW

File: sdk/spring/azure-spring-boot/pom.xml [hunk 1]
```diff
@@ -326,8 +327,9 @@
                 <include>com.microsoft.azure:azure-servicebus-jms:[0.0.7]</include> <!-- {x-include-update;com.microsoft.azure:azure-servicebus-jms;external_dependency} -->
                 <include>com.github.spotbugs:spotbugs-annotations:[4.2.0]</include> <!-- {x-include-update;com.github.spotbugs:spotbugs-annotations;external_dependency} -->
                 <include>com.fasterxml.jackson.core:jackson-databind:[2.12.2]</include> <!-- {x-include-update;com.fasterxml.jackson.core:jackson-databind;external_dependency} -->
-                <include>net.minidev:json-smart:[2.3]</include> <!-- {x-include-update;net.minidev:json-smart;external_dependency} -->
-                <include>com.microsoft.azure:msal4j:[1.9.1]</include> <!-- {x-include-update;com.microsoft.azure:msal4j;external_dependency} -->
+                <!-- TODO: add {x-include-update;net.minidev:json-smart;external_dependency} this tag below as part of PR: https://github.com/Azure/azure-sdk-for-java/pull/20770-->
```
Comment (g2vinay) [inline sdk/spring/azure-spring-boot/pom.xml:330]: DIsabling this update.
As spring team aims to update this via this PR (#20770) post their releases.

File: sdk/identity/azure-identity/pom.xml [hunk 2]
```diff
@@ -116,12 +110,12 @@
           <rules>
             <bannedDependencies>
               <includes>
-                <include>com.microsoft.azure:msal4j:[1.9.1]</include> <!-- {x-include-update;com.microsoft.azure:msal4j;external_dependency} -->
+                <include>com.microsoft.azure:msal4j:[1.10.0]</include> <!-- {x-include-update;com.microsoft.azure:msal4j;external_dependency} -->
                 <include>com.microsoft.azure:msal4j-persistence-extension:[1.1.0]</include> <!-- {x-include-update;com.microsoft.azure:msal4j-persistence-extension;external_dependency} -->
                 <include>com.nimbusds:oauth2-oidc-sdk:[8.36.1]</include> <!-- {x-include-update;com.nimbusds:oauth2-oidc-sdk;external_dependency} -->
```
Comment (g2vinay) [inline sdk/identity/azure-identity/pom.xml:115]: remove oauth dep from here.

File: sdk/spring/azure-spring-boot/pom.xml [hunk 3]
```diff
@@ -224,7 +224,8 @@
     <dependency>
       <groupId>net.minidev</groupId>
       <artifactId>json-smart</artifactId>
-      <version>2.3</version> <!-- {x-version-update;net.minidev:json-smart;external_dependency} -->
+      <!-- TODO: add {x-version-update;net.minidev:json-smart;external_dependency} this version tag below as part of PR: https://github.com/Azure/azure-sdk-for-java/pull/20770-->
```
Comment (g2vinay) [inline sdk/spring/azure-spring-boot/pom.xml:227]: DIsabling this update.
As spring team aims to update this via this PR (#20770) post their releases.

**Review by jianghaolu — APPROVED** [review]

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**